src/tests/scenarios/test_iso_yielder_setting.cairo

```diff
@@ -0,0 +1,731 @@
+    let prices: Array<u256> = array![0, price];
+
+    farmer.set_prices(times.span(), prices.span());
+    // Owner approve yielder to spend his tokens
+    let project = IERC721Dispatcher { contract_address: project };
+    project.set_approval_for_all(yielder, true);
+    // Anyone approve yielder to spend his tokens
+    set_contract_address(*signers.anyone);
+    project.set_approval_for_all(yielder, true);
+}
+
+fn setup_minter(project: ContractAddress, minter: ContractAddress, signers: @Signers) {
+    let project = IERC721Dispatcher { contract_address: project };
+    // Owner approve minter to spend his tokens
+    set_contract_address(*signers.owner);
+    project.set_approval_for_all(minter, true);
+    // Anyone approve minter to spend his tokens
+    set_contract_address(*signers.anyone);
+    project.set_approval_for_all(minter, true);
+}
+
+fn setup(price: u256) -> (Signers, Contracts) {
+    // Deploy
+    let signers = Signers { owner: deploy_account('OWNER'), anyone: deploy_account('ANYONE'), anyone2: deploy_account('ANYONE2') };
+    let project = deploy_project(signers.owner);
+    let erc20 = deploy_erc20(signers.owner);
+    let yielder = deploy_yielder(project, erc20, signers.owner);
+    let minter = deploy_minter(project, erc20, signers.owner);
+
+    // Setup
+    setup_project(project, @signers);
+    setup_erc20(erc20, yielder, @signers);
+    setup_yielder(project, erc20, yielder, @signers, price);
+    setup_minter(project, minter, @signers);
+
+    // Setup Minter 
+    set_contract_address(signers.owner);
+    let project_minter = IMinterDispatcher { contract_address: project };
+    project_minter.add_minter(SLOT, minter);
+
+    // Return
+    let contracts = Contracts { project: project, erc20: erc20, yielder: yielder, minter: minter };
+    (signers, contracts)
+}
+
+fn setup_for_apr(n: u64) -> (Signers, Contracts) {
+    // Deploy
+    let signers = Signers { owner: deploy_account('OWNER'), anyone: deploy_account('ANYONE'), anyone2: deploy_account('ANYONE2'), };
+    let project = deploy_project(signers.owner);
+    let erc20 = deploy_erc20(signers.owner);
+    let yielder = deploy_yielder(project, erc20, signers.owner);
+    let minter = deploy_minter(project, erc20, signers.owner);
+
+    // Setup
+    setup_project_apr(project, @signers, n);
+    setup_erc20(erc20, yielder, @signers);
+    // setup_yielder(project, erc20, yielder, @signers);
+    setup_minter(project, minter, @signers);
+
+    // Setup Minter 
+    set_contract_address(signers.owner);
+    let project_minter = IMinterDispatcher { contract_address: project };
+    project_minter.add_minter(SLOT, minter);
+
+    // Return
+    let contracts = Contracts { project: project, erc20: erc20, yielder: yielder, minter: minter };
+    (signers, contracts)
+}
+
+#[test]
+#[available_gas(400_000_000)]
+fn test_yielder_iso_banegas_farm_deposit_withdraw_value() {
```

Comment:
Which scenario from [#289](https://github.com/Carbonable/carbon-protocol/issues/289) is this test referring to?

Reply:
removed